service/src/main.rs

```diff
@@ -677,38 +673,32 @@ fn execute_trusted_calls<E: Sidechain>(enclave_api: &E) {
 }
 
 /// gets a list of blocks that need to be synced, ordered from oldest to most recent header
-/// blocks that need to be synced are all blocks from the current header to the last synced header, iterating over parent
-fn get_blocks_to_sync(
+/// starting from the last synced head until the current head but at most BLOCK_SYNC_BATCH_SIZE.
+fn get_block_chunk_to_sync(
 	api: &Api<sr25519::Pair, WsRpcClient>,
 	last_synced_head: &Header,
 	curr_head: &SignedBlock,
 ) -> Vec<SignedBlock> {
 	let mut blocks_to_sync = Vec::<SignedBlock>::new();
```

Comment:
This has obviously been like that before, but I would like to see an improvement here.

1. Please add this code to the api-client-extensions crate: https://github.com/integritee-network/worker/blob/master/core-primitives/api-client-extensions/src/chain.rs. This will also help us with mocking, as we can mock that trait and returned pre-defined blocks from the method then.
2. If I am not mistaken, we only need the numbers of `last_synced_head` and `curr_head`. I think the code will be much more readable if we don't pass the entire headers but only the numbers.

I think we can improve the overall readability of the code if we introduce the following api method: `api.get_blocks(last_synced_header_num, last_synced_header_num + BLOCKS_CHUNK_SIZE)`, so we don't need to check anything against the chunk size inside this method. What do you think?